third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -1118,6 +1127,15 @@ struct Frame {
     function: Arc<Function>,
     ty_args: Vec<Type>,
     local_tys: Vec<Type>,
+    ty_cache: FrameCache,
+}
+
+#[derive(Default)]
+struct FrameCache {
+    struct_field_instantiation: HashMap<StructDefInstantiationIndex, Vec<Type>>,
+    struct_def_instantiation: HashMap<StructDefInstantiationIndex, Type>,
+    field_instantiation: HashMap<FieldInstantiationIndex, Type>,
+    single_sig_token: HashMap<SignatureIndex, Type>,
```

Comment:
From the previous incidents we know that HashMap is non-deterministic. Wondering if we should use it here... It does feel harmless in this case though.

BTW if you have the spare bandwidth, it might be helpful to run some simple benchmarks and see if HashMap is indeed faster than BTreeMap.